scipy/optimize/minpack.py

```diff
@@ -145,12 +144,19 @@ def fsolve(func, x0, args=(), fprime=None, full_output=0,
         info['fvec'] = res['fun']
         return x, info, res['status'], res['message']
     else:
+        status = res['status']
+        msg = res['message']
+        if status != 1:
+            if status in [2, 3, 4, 5]:
+                warnings.warn(msg, RuntimeWarning)
+            else:
+                raise TypeError(msg)
```

Comment:
What do you think of doing this:
```python
if status == 0:
    raise TypeError(msg)
elif status == 1:
    pass
else:
    warnings.warn(msg, RuntimeWarning)
```
That way there won't be a warning if everything goes okay.